### PR TITLE
🌱 Improve KCP conditions when node status in unknown

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR improve Machine/KCP conditions when the node status is unknown.

**Which issue(s) this PR fixes**:
Fixes https://github.com/kubernetes-sigs/cluster-api/issues/4562

### CODE REVIEW

File: controlplane/kubeadm/internal/workload_cluster_conditions.go [hunk 1]
```diff
@@ -19,9 +19,10 @@ package internal
 import (
 	"context"
 	"fmt"
-	"sigs.k8s.io/cluster-api/util/collections"
 	"strings"
 
+	"sigs.k8s.io/cluster-api/util/collections"
+
```
Comment (vincepri) [inline controlplane/kubeadm/internal/workload_cluster_conditions.go:25]:
```suggestion
```

**Review by vincepri — COMMENTED** [review]
/lgtm

**Review by CecileRobertMichon — COMMENTED** [review]
lgtm

File: controlplane/kubeadm/internal/workload_cluster_conditions.go [hunk 2]
```diff
@@ -362,6 +362,13 @@ func (w *Workload) updateStaticPodCondition(ctx context.Context, machine *cluste
 		Name:      staticPodName(component, node.Name),
 	}
 
+	// If node ready is unknown there is a good chance that kubelet is not updating mirror pods, so we consider pod status
```
Comment (enxebre) [inline controlplane/kubeadm/internal/workload_cluster_conditions.go:365]: nit: this block could go before declaring `podKey` block.

Comment (fabriziopandini) [reply controlplane/kubeadm/internal/workload_cluster_conditions.go:365]: done!

**Review by vincepri — APPROVED** [review]
/approve
/lgtm

**Pull request merged**